modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java

```diff
@@ -561,7 +574,8 @@ private boolean initme() {
             }
         }
         userName = (String) parameters.get(USERNAME);
-        password = (String) parameters.get(PASSWORD);
+        password = resolveSecureVaultExpressions((String) parameters.get(PASSWORD));
```

Comment:
There is no problem, if password parameter is not encrypted. resolveSecureVaultExpressions method can handle it by return the actual password.